core/node/proof_data_handler/src/lib.rs

```diff
@@ -48,16 +84,38 @@ pub async fn run_server(
             ),
         );
 
-    axum::Server::bind(&bind_address)
-        .serve(app.into_make_service())
-        .with_graceful_shutdown(async move {
-            if stop_receiver.changed().await.is_err() {
-                tracing::warn!("Stop signal sender for proof data handler server was dropped without sending a signal");
-            }
-            tracing::info!("Stop signal received, proof data handler server is shutting down");
-        })
-        .await
-        .context("Proof data handler server failed")?;
-    tracing::info!("Proof data handler server shut down");
-    Ok(())
+    if config.tee_support {
+        return router.route(
```

Comment:
Probably a more idiomatic way would be to make `router` mutable and here simply modify it (e.g. `router = router.route(..)`), so that we have a single return point. The benefit here would be that this approach would allow the conditional composition of several modules out of box.

Reply:
[Fixed](https://github.com/matter-labs/zksync-era/pull/1993/commits/c4e4074a7135e3ff391ef0a315b939d079a13a0b). ✅